### PR TITLE
Add Queue implementation

### PR DESCRIPTION
**Pull request by radusqrt**
Closes https://github.com/NITSkmOS/Algorithms/issues/58

### CODE REVIEW

File: data_structures/Queue/Queue.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    void pop() {
+        if (empty()) return;
+        queue.pop_back();
+    }
+
+    std::experimental::optional<T> top() {
+        if (empty()) return std::experimental::nullopt;
+        return queue.back();
+    }
+};
```
Comment (sangamcse) [inline data_structures/Queue/Queue.h]: EOL is missing here. Rest the code LGTM 😄

File: data_structures/Queue/main.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    q.push(3);
+    assert(!q.empty());
+    assert(q.top().value_or(-1) == 2);
+    q.pop();
+    assert(q.top().value_or(-1) == 3);
+    q.pop();
+    assert(q.empty());
+    assert(q.top().value_or(-1) == -1);
+    return 0;
+}
```
Comment (sangamcse) [inline data_structures/Queue/main.cpp]: EOL here, too.

**Comment by radusqrt** [comment]
Hopefully, everything is done now. Let me know.

**Comment by sangamcse** [comment]
@radusqrt, that's very kind of you.
One thing just went wrong and that is you just created 3 commits. Please squash them into one.

For guide https://makandracards.com/makandra/527-squash-several-git-commits-into-a-single-commit

**Comment by radusqrt** [comment]
I got a bit tangled and created a new PR. You can close this one and go to: https://github.com/NITSkmOS/Algorithms/pull/71

**Comment by sangamcse** [comment]
moving to https://github.com/NITSkmOS/Algorithms/pull/71

**Pull request closed without merging**